arangod/Scheduler/SupervisedScheduler.h

```diff
@@ -138,18 +143,27 @@ class SupervisedScheduler final : public Scheduler {
 
   // Since the lockfree queue can only handle PODs, one has to wrap lambdas
   // in a container class and store pointers. -- Maybe there is a better way?
-  boost::lockfree::queue<WorkItem*> _queues[3];
+  boost::lockfree::queue<WorkItem*> _queues[NumberOfQueues];
 
   // aligning required to prevent false sharing - assumes cache line size is 64
   alignas(64) std::atomic<uint64_t> _jobsSubmitted;
   alignas(64) std::atomic<uint64_t> _jobsDequeued;
   alignas(64) std::atomic<uint64_t> _jobsDone;
 
-  uint64_t const _minNumWorkers;
-  uint64_t const _maxNumWorkers;
-  uint64_t const _maxFifoSize;
-  uint64_t const _fifo1Size;
-  uint64_t const _fifo2Size;
+  size_t const _minNumWorker;
+  size_t const _maxNumWorker;
+  uint64_t const _maxFifoSizes[NumberOfQueues];
+  size_t const _ongoingLowPriorityLimit;
+
+  // During a queue operation there a two reasons to manually wake up a worker
+  //  1. the queue length is bigger than _wakeupQueueLength and the last submit time
+  //      is bigger than _wakeupTime_ns.
+  //  2. the last submit time is bigger than _definitiveWakeupTime_ns.
+  //
+  // The last submit time is a thread local variable that stores the time of the last
+  // queue operation.
+  alignas(64) std::atomic<uint64_t> _wakeupQueueLength;            // q1
+  std::atomic<uint64_t> _wakeupTime_ns, _definitiveWakeupTime_ns;  // t3, t4
```

Comment:
Wondering why these variables are reintroduced in this PR.
I deleted them previously, because they did nothing. :thinking: